iroh-p2p/src/rpc.rs

```diff
@@ -394,6 +391,54 @@ impl P2p {
         Ok(peer_info_from_lookup(lookup))
     }
 
+    #[tracing::instrument(skip(self))]
+    fn gossipsub_subscribe(self, req: GossipsubSubscribeRequest) -> GossipsubEventStream {
+        async move {
+            self.gossipsub_subscribe_0(req)
+                .await
+                .expect("FIX THIS - NEEDS TO BE ABLE TO RETURN A RESULT")
+        }
+        .flatten_stream()
+        .boxed()
+    }
```

Comment:
@rklaehn I'm coming into conflict with this method in two ways.

For context, a `GossipsubEventStream` is `BoxStream<'static, Box<GossipsubSubscribeResponse>>`

First question/problem: I can't figure out how to handle the possible error from `gossipsub_subscribe_0` When I try to use `?` instead of `expect` here, I keep running into 
```
cannot use the `?` operator in an async block that returns `Pin<Box<dyn Stream<Item = Box<iroh_rpc_types::p2p::GossipsubSubscribeResponse>> + std::marker::Send>>`
    |                    this function should return `Result` or `Option` to accept `?
```
And when I looked up how to solve this, I didn't understand the syntax that the solutions suggested. Would love some help!

The second conflict I'm running into is with my "preferred" return type. My "dream" scenario is that the p2p RpcClient's `gossipsub_subscribe` can have a return type of:
`Result<impl Stream<Item=GossipsubEvent>>`

But, it seems like I'm forced to return a `Result<GossipsubEvent>` Item, based on the expectations of `server_streaming`. This isn't critical, but It would be nice to streamline the API in this way.

Reply:
A server streaming interaction returns a stream, not a result of stream. So if you have a situation where you have a possible error on creation of the stream, but then no errors once the stream is created, you have no choice but to have the stream item be a `Result`. Otherwise the try_flatten_stream thing won't work.

So the result must be `impl Stream<Item=Result<GossipsubEvent>>`, or we must extend quic-rpc to support a result of stream. Since it is all just decorators and most of the critical work is in the transports, this should be possible. Just a bit of work, so the question is how strongly you feel about this.

Is it really the case that once you have subscribed there is nothing whatsoever that can go wrong? That is usually quite rare.